guiprep.pl

```diff
@@ -231,170 +235,178 @@
 my $main = MainWindow->new(
 	-title =>		'Guiprep Pre-processing Toolkit - '.$currentver,
 );
+my $int = $main->interp;
 
 # Set window size (pixels) the script TRIES to open with.
 $main->geometry($geometry);
 
 # Set the minimum window size.
-$main->minsize(qw(600 480));
+$main->minsize(qw(1024 768));
 
-# Set an interesting palette color combo.
-$main->setPalette("$palette");
+# Set an interesting theme color combo.
+$main->ttkSetTheme("$theme");
 
-my $mainbframe = $main->Frame()->pack(-anchor => 'nw');
+my $mainbframe = $main->ttkFrame()->pack(-anchor => 'nw');
 
 my $bottomlabel = $mainbframe->Label(
 	-text => "Working from the $pwd directory.",
-	-font => '-*-Helvetica-Medium-R-Normal--*-160-*-*-*-*-*-*'
 )->pack(-anchor => 'nw');
 
-my $maintopframe = $main->Frame(
+my $maintopframe = $main->ttkFrame(
 )->pack(-anchor => 'n', -expand => 'y', -fill => 'both');
 
-my $book = $maintopframe->NoteBook( -ipadx => 6, -ipady => 6);
-
-my $page2 = $book->add("page2", -label => "Select Options", -raisecmd => sub {$interrupt = 1;chdir $pwd;});
-my $page1 = $book->add("page1", -label => "Process Text", -raisecmd => \&updateblist);
-my $page8 = $book->add("page8", -label => "Search", -raisecmd => sub{$interrupt = 1; searchclear(); $filesearchindex1 = 1;
-									@searchfilelist=(); $main->update; $main->Busy; searchincdec(); $main->Unbusy;});
-my $page3 = $book->add("page3", -label => "Headers & Footers", -raisecmd => sub {emptybox(); $interrupt = 1;chdir $pwd;});
-my $page4 = $book->add("page4", -label => "Change Directory", -raisecmd => sub {$interrupt = 1;chdir $pwd;});
-my $page6 = $book->add("page6", -label => "Program Prefs", -raisecmd => sub {$interrupt = 1;chdir $pwd;});
-my $page5 = $book->add("page5", -label => "About", -raisecmd => sub {$interrupt = 1;chdir $pwd;});
+my $book = $maintopframe->ttkNotebook();# -ipadx => 6, -ipady => 6);
+
+$main->bind('<<NotebookTabChanged>>',
+	sub{
+		for ($book->tab($book->select())->[7]) {
+				if ('Select Options') { sub {print "Select Options";$interrupt = 1;chdir $pwd;} }
+				elsif ('Process Text') { \&updateblist }
+		        elsif ('Search') { sub{$interrupt = 1; searchclear(); $filesearchindex1 = 1;
+									@searchfilelist=(); $main->update; $main->Busy; searchincdec(); $main->Unbusy;} }
+			    elsif ('Headers & Footers') { sub {emptybox(); $interrupt = 1;chdir $pwd;} }
+                elsif ('Change Directory') { sub {$interrupt = 1;chdir $pwd;} }
+                elsif ('Program Prefs') { sub {$interrupt = 1;chdir $pwd;} }
+                elsif ('About') { sub {$interrupt = 1;chdir $pwd;} }
+                else { sub {$interrupt = 1;chdir $pwd;} }
+		}});
+
+my $page2 = $book->ttkFrame();
+my $page1 = $book->ttkFrame();
+my $page8 = $book->ttkFrame();
+my $page3 = $book->ttkFrame();
+my $page4 = $book->ttkFrame();
+my $page6 = $book->ttkFrame();
+my $page5 = $book->ttkFrame();
+
+my $page2c = $book->add($page2, -text => "Select Options");
```

Comment:
Do we need to assign these to anything if the variables themselves aren't used? Can we just call the `$book->add()` methods instead?

These would probably make more sense if we put them next to the page definitions above.

And while we're here maybe just order them by the page number (I don't know why they are out of order here -- are they assembled in a different order in the widget?).

Reply:
Totally true, I think I made them variables before what I was sure would be needed elsewhere. 100% works without declaring them.

The page order comes in the order we've declared them, which differs from the numeric order. It would be possible to make what's currently `$page2` be `$page1`, but then all the `$p1foo` stuff and `$p2foo` stuff would all need to be reorganized accordingly. That seems like a totally reasonable thing to refactor eventually, but seems like it should happen separately from what's already a fairly extensive change. Does that seem right, or do you want to pull the bandaid all at once?

Reply:
Agreed we shouldn't rename anything in this PR. Can we add a comment that these are in the order they are in the UI which differs from the numeric order for the next person who find this odd?

Reply:
Fixed with ad76a9b.